File: javascript/httpQuery.js
import http from 'node:https';

// vanilla query node 16

export default function httpQuery() {


    http.get('https://example.com/', (res) => {
        const {statusCode} = res;
        console.log('Request com status code ', statusCode);
        const data = [];

        res.on('data', block => data.push(block))
        res.on('end', () => {
            console.log('fin do request');
            console.log(Buffer.concat(data).toString())
        })
    }).on('error', (err) => {
        console.log('Erro');
        console.log({err})
    })

}

File: python/http_query.py
import urllib.request
import urllib.error as web_error


url = 'https://example.com/'
try:
    response = urllib.request.urlopen(url)
    print(f'Status code {response.status}')
    print(response.read().decode('utf-8'))
except web_error.HTTPError as err:
    print(f'Status code {err.code}')
    print(err.reason)
    print(err.headers)


